src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -210,19 +208,25 @@ public int[] getRequiredJavadocTokens() {
 
     @Override
     public void visitJavadocToken(DetailNode ast) {
-        final Optional<DetailNode> inlineTag = getInlineTagNode(ast);
-        final DetailNode inlineTagNode = inlineTag.orElse(null);
-        if (inlineTag.isPresent()
-            && isSummaryTag(inlineTagNode)
-            && isDefinedFirst(inlineTagNode)) {
-            validateSummaryTag(inlineTagNode);
-        }
-        else if (inlineTag.isPresent() && isInlineReturnTag(inlineTagNode)) {
-            validateInlineReturnTag(inlineTagNode);
-        }
-        else if (!startsWithInheritDoc(ast)) {
+        final Optional<DetailNode> inlineTagNode = getInlineTagNode(ast);
+        final AtomicBoolean shouldValidateUntaggedSummary = new AtomicBoolean(true);
+
+        inlineTagNode.ifPresent(node -> {
```

Comment:
can we do 
```
.map(some function)
.orElse(some function);
```

Reply:
tried something like : 

        final boolean shouldValidateUntaggedSummary = inlineTagNode
                .map(this::validateInlineTagNode)
                .orElse(false);

        if (shouldValidateUntaggedSummary && !startsWithInheritDoc(ast)) {
            validateUntaggedSummary(ast);
        }
        
and got `NAB_NEEDLESS_BOOLEAN_CONSTANT_CONVERSION` for `.orElse(false);`

Reply:
how about this approach : 
```
public void visitJavadocToken(DetailNode ast) {
    final Optional<DetailNode> inlineTagNode = getInlineTagNode(ast);
    AtomicBoolean shouldValidateUntaggedSummary = new AtomicBoolean(true);

    inlineTagNode.ifPresent(node -> {
         shouldValidateUntaggedSummary.set(validateInlineTagNode(node));
    });
      if (shouldValidateUntaggedSummary.get() && !startsWithInheritDoc(ast)) {
            validateUntaggedSummary(ast);
        }

    }
```
can't do without AtomicBoolean .